src/i965_gpe_utils.c

```diff
@@ -1406,6 +1406,27 @@ i965_object_surface_to_2d_gpe_resource(struct i965_gpe_resource *res,
 }
 
 void
+i965_object_surface_to_2d_gpe_resource_with_align(struct i965_gpe_resource *res,
+                                                  struct object_surface *obj_surface)
+{
+    unsigned int swizzle;
```

Comment:
i965_object_surface_to_2d_gpe_resource() and i965_object_surface_to_2d_gpe_resource_with_align are almost the same. I suggest take the alignment as an input parameter to i965_object_surface_to_2d_gpe_resource_with_align(), define  i965_object_surface_to_2d_gpe_resource as below
void i965_object_surface_to_2d_gpe_resource(struct i965_gpe_context *gpe_context,
                             struct i965_gpe_surface *gpe_surface,
                             int index)
{
      i965_object_surface_to_2d_gpe_resource_with_align(gpe_context, gpe_surface, 1);
}

We will only modify one function if there is any error in i965_object_surface_to_2d_gpe_resource(),i965_object_surface_to_2d_gpe_resource_with_align(). In addition, we can set other alignments if needed.